core/app/[locale]/(default)/webpages/normal/[id]/page.tsx

```diff
@@ -1,26 +1,52 @@
 import type { Metadata } from 'next';
 import { notFound } from 'next/navigation';
 
+import { Breadcrumb } from '@/vibes/soul/primitives/breadcrumbs';
+import { WebPageContent, WebPage as WebPageData } from '@/vibes/soul/sections/web-page-content';
+
 import { getWebpageData } from './page-data';
 
 interface Props {
   params: Promise<{ id: string }>;
 }
 
-export async function generateMetadata({ params }: Props): Promise<Metadata> {
-  const { id } = await params;
-
+async function getWebPage(id: string): Promise<WebPageData> {
```

Comment:
We need to wrap this in `cache` to memoize the response.

Reply:
@chanceaclark The `getWebpageData({ id: decodeURIComponent(id) })` function is wrapped in `cache` in the `page-data.ts` file 👍

Reply:
Why do we have two fns to fetch the data then. Can we make it just one function to fetch the data? I'm not sure we should be calling `notFound()` within `generateMetadata`.

Reply:
@chanceaclark I noticed this is a similar pattern we follow in most other pages/components, the `getWebPage` function is responsible for making the GQL call and then converting it into an object ready to be passed into the vibes component as the promise. 

I can update it to not return `notFound` though 👍  My line of thinking there was that if the webpage isn't found, then returning `notFound` should trump everything else. If the page isn't found, why would we want to do anything with the metadata?

Reply:
I believe this is the pattern. @chanceaclark